build_runner_core/test/generate/build_test.dart

```diff
@@ -1131,6 +1131,35 @@ void main() {
           },
           writer: writer);
     });
+
+    test('no implicit dependency on primary input contents', () async {
+      var builders = [applyToRoot(SiblingCopyBuilder())];
```

Comment:
[nit] We moved away from implementing these behaviors as individual classes.

```dart
applyToRoot(TestBuilder(buildExtensions: appendExtension('.new'),
    build: (step, _) async {
      var sibling = ...
    }))
```

Reply:
I can't use that because it uses `canRead` always and you can't override it

Reply:
ah... I'm sure I had a good reason for that... 😬